oiosaml/src/main/java/dk/gov/oio/saml/servlet/LogoutRequestHandler.java

```diff
@@ -113,4 +113,41 @@ public void handleGet(HttpServletRequest httpServletRequest, HttpServletResponse
     public void handlePost(HttpServletRequest httpServletRequest, HttpServletResponse httpServletResponse) throws ExternalException, InternalException, IOException {
         handleGet(httpServletRequest, httpServletResponse);
     }
+
+    @Override
+    public void handleSOAP(HttpServletRequest httpServletRequest, HttpServletResponse httpServletResponse) throws ExternalException, InternalException, IOException {
+        log.debug("Handling SOAP LogoutRequest");
+
+        //TODO: When updated from NLRFIM-109 - refactor class so that biz logic is separated from handling the protocol
+
+        // IdP Initiated, generate response
+        MessageContext<SAMLObject> context = decodeSOAP(httpServletRequest);
+        LogoutRequest logoutRequest = getSamlObject(context, LogoutRequest.class);
+
+        // TODO log as above + debug
+        log.info("Incoming LogoutRequest - ID:'{}' Issuer:'{}' IssueInstant:'{}' SessionIndexes:'{}' Destination:'{}'");
+
+        // Validate logout request, we log the user out not matter what, but we should log if the request is wrong
+        
+        // Delete session
+        httpServletRequest.getSession().invalidate();
```

Comment:
This is not the correct session and therefore the log of the actual session which needs to be terminated is still active. Hence the implementation will have to be changed the external session storage implemented. Don't think it is a good idea to merge it now, other an addition to the NLRFIM-121 must be made.

Reply:
I agree